### PR TITLE
docs: add a note on result size and containers

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This adds a small note on how Results work and how the number of steps
can affect the maximum size of the Task results.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind docs

/cc @abayer @afrittoli @jerop @lbernick 

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in
(if there are no user facing changes, use release note "NONE")

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/tasks.md [hunk 1]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
 It's important to note that Tekton does not perform any processing on the contents of results; they are emitted
 verbatim from your Task including any leading or trailing whitespace characters. Make sure to write only the
 precise string you want returned from your `Task` into the `/tekton/results/` files that your `Task` creates.
-You can use [`$(results.name.path)`](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-task)
+You can use [`$(results.name.path)`](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-task**
```
Comment (abayer) [inline docs/tasks.md:648]: Missing closing paren?

File: docs/tasks.md [hunk 2]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
...
 to avoid having to hardcode this path.
 
+Note: Tekton uses [termination
+messages](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#writing-and-reading-a-termination-message). As
+written in
+[tektoncd/pipeline#4808](https://github.com/tektoncd/pipeline/issues/4808),
+This has some *shortcommings*. The main point here is the more
```
Comment (abayer) [inline docs/tasks.md:655]: This should be “shortcomings” with one m.

File: docs/tasks.md [hunk 3]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
...
+container we have in our pod, *the smaller the size of one container's
```
Comment (abayer) [inline docs/tasks.md:656]: This section doesn’t read right - I’ll get you a rewording in an hour or so. =)

**Review by abayer — CHANGES_REQUESTED** [review]
There are some typos and wording issues.

File: docs/tasks.md (hunk 3 above)
Comment (abayer) [reply docs/tasks.md:656]: Suggested replacement for this paragraph:
```
Note: Tekton uses [termination
messages](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#writing-and-reading-a-termination-message). As
written in
[tektoncd/pipeline#4808](https://github.com/tektoncd/pipeline/issues/4808),
this has some *shortcommngs*. The main point here is that the more
containers we have in our pod, *the smaller the allowed size of each container's
message*, which translates in Tekton tothe **more steps you
have in a Task, the smaller the result for each step can be**. Be aware that the number of
steps in a Task affects the maximum size of a Result. *For example, if
you have 10 steps, the size of each step's Result will have a a maximum of less than 1KB*.
```

File: docs/tasks.md [hunk 4]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
...
 precise string you want returned from your `Task` into the `/tekton/results/` files that your `Task` creates.
-You can use [`$(results.name.path)`](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-task)
+You can use [`$(results.name.path)`](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#variables-available-in-a-task)**
 to avoid having to hardcode this path.
 
+Note: Tekton uses [termination
+messages](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#writing-and-reading-a-termination-message). As
+written in
+[tektoncd/pipeline#4808](https://github.com/tektoncd/pipeline/issues/4808),
+this has some *shortcommngs*. The main point here is that the more
```
Comment (lbernick) [inline docs/tasks.md:655]: typo here: shortcomings

File: docs/tasks.md [hunk 5]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
...
+containers we have in our pod, *the smaller the allowed size of each container's
+message*, which translates in Tekton tothe **more steps you
```
Comment (lbernick) [inline docs/tasks.md:657]: missing space: tothe -> to the

File: docs/tasks.md [hunk 6]
```diff
@@ -645,9 +645,20 @@ a `results` field but it's the responsibility of the `Task` to generate its cont
...
+have in a Task, the smaller the result for each step can be**. Be aware that the number of
+steps in a Task affects the maximum size of a Result. *For example, if
+you have 10 steps, the size of each step's Result will have a a maximum of less than 1KB*.
```
Comment (lbernick) [inline docs/tasks.md:660]: have a a maximum -> have a maximum

**Comment by lbernick** [comment]
/hold

just want to make sure typos are addressed before this is merged

**Comment by vdemeester** [comment]
/retest
network issue on the `build-tests` pod.. 😓

**Pull request merged**